MovieQuiz/Presentation/MovieQuizPresenter.swift

```diff
@@ -0,0 +1,124 @@
+import Foundation
+import UIKit
+
+
+final class MovieQuizPresenter: QuestionFactoryDelegate {
+    let questionsAmount = 10
+    var score = 0
+    var currentQuestion: QuizQuestion?
+    private var currentQuestionIndex = 0
+
+    var questionFactory: QuestionFactoryProtocol?
+    private var viewController: MovieQuizViewControllerProtocol?
+    private var statisticService: StatisticService
+
+    init(viewController: MovieQuizViewControllerProtocol) {
+        self.viewController = viewController
+        statisticService = StatisticServiceImplementation()
+        questionFactory = QuestionFactory(moviesLoader: MoviesLoader(), delegate: self)
+        questionFactory?.loadData()
+        viewController.showLoadingIndicator()
+    }
+
+    func isLastQuestion() -> Bool {
+        currentQuestionIndex == questionsAmount - 1
+    }
+
+    func switchToNextQuestion() {
+        currentQuestionIndex += 1
+    }
+
+    func convert(model: QuizQuestion) -> QuizStepViewModel {
+        let image = UIImage(data: model.image)
+        let step = QuizStepViewModel(
+            image: image,
+            questionText: model.question,
+            questionNumber: "\(currentQuestionIndex + 1)/\(questionsAmount)"
+        )
+        return step
+    }
+
+    private func didAnswer(isYes: Bool) {
+        guard let currentQuestion = currentQuestion else {
+            return
+        }
+        let givenAnswer = isYes
+        let answerWasCorrect = givenAnswer == currentQuestion.answer
+        if answerWasCorrect {
+            score += 1
+        }
+        viewController?.toggleAnswerButtons()
+        viewController?.setImageBorder(answerWasCorrect: answerWasCorrect)
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
+            [weak self] in
```

Comment:
```suggestion
        DispatchQueue.main.asyncAfter(deadline: .now() + 1) { [weak self] in
```

принято оставлять список захвата на той же строке, где начинается замыкание